h/static/scripts/base/upgrade-elements.js

```diff
@@ -1,6 +1,18 @@
 'use strict';
 
 /**
+ * Mark an element as having been upgraded.
+ */
+function markReady(element) {
+  var HIDE_CLASS = 'u-js-hide-when-loading';
+  var hideOnLoad = Array.from(element.querySelectorAll('.' + HIDE_CLASS));
```

Comment:
> Doesn't this break your own rules about which classes it's ok to reference from JavaScript?

This is a bit unusual in that we _do_ style this class, _do_ want to use it on its own (so not in the context of a specific component) and also want to reference it from JS. Probably the least ugly way to break the existing rules is to make this a state class (`is-hidden-when-loading`) and allow it to be used along with any element.

Another thought I had was to create a custom wrapper element:
```
<hide-when-loading>
... content ...
</hide-when-loading>
```